groovy-language-server/src/main/java/com/palantir/ls/util/Ranges.java

```diff
@@ -85,4 +85,32 @@ public static boolean contains(Range range, Position position) {
                 && POSITION_COMPARATOR.compare(range.getEnd(), position) >= 0;
     }
 
+    /**
+     * Returns the maximum position.
+     */
+    public static Position max(Position position1, Position position2) {
+        return POSITION_COMPARATOR.reversed().compare(position1, position2) < 0 ? position1 : position2;
+    }
+
+    /**
+     * Returns the minimum position.
+     */
+    public static Position min(Position position1, Position position2) {
+        return POSITION_COMPARATOR.compare(position1, position2) < 0 ? position1 : position2;
+    }
+
+    /**
+     * Returns whether range1 and range2 intersect. Assumes that a range is inclusive on its start and exclusive on its
+     * end, which means that if range1 ends on the same position as range2 starts, this is not considered intersecting.
+     */
+    public static boolean intersects(Range range1, Range range2) {
+        checkArgument(isValid(range1), String.format("range1 is not valid: %s", range1.toString()));
+        checkArgument(isValid(range2), String.format("range2 is not valid: %s", range2.toString()));
+
+        Position maxStart = max(range1.getStart(), range2.getStart());
```

Comment:
only need to check (1) if the range1.start == range1.end, check range2.start > range.end (2) otherwise check range2.start >= range1.end